foxpuppet/windows.py

```diff
@@ -43,3 +47,28 @@ def window_element(self):
         """
 
         return self.selenium.find_element(By.CSS_SELECTOR, ':root')
+
+    def close(self, handle):
+        """ Closes the window with the specified handle """
+
+        self.switch_to(handle)
+
+        self.selenium.close()
+
+    def switch_to(self, target):
+        """Switches context to the specified chrome window"""
+
+        target_handle = None
+
+        # Check if requested handle is available
+        if target in self.all:
+            target_handle = target
+        else:
+            print("Handle not found")
+
+        if not target_handle:
+            self.selenium.switch_to.window(self.current)
+
+        # Switch to window if not currently on it
+        if target_handle != self.current:
+            self.selenium.switch_to.window(target_handle)
```

Comment:
I don't know why a user would try switching to the current window, but it shouldn't cause any problems. I wonder if we're writing unnecessary code here. A user (or FoxPuppet) can just call `self.selenium.switch_to.window(handle)` directly and allow any exceptions to be thrown. I don't feel like we're gaining anything be abstracting this call.